src/components/Nav/NavList/NavList.js

```diff
@@ -116,6 +116,15 @@ export default class NavList extends Component {
         role="menuitem"
       >
         <div className={`${navItemNamespace}__link`}>
+          <div>
+            {this.props.hasIcon && (
+              <img
```

Comment:
I believe there is an `<Icon/>` element that is used in the library for this sort of use case.